layers/core_validation.cpp

```
@@ -1664,6 +1693,70 @@ static bool ValidatePipelineUnlocked(layer_data *dev_data, std::vector<std::uniq
         }
     }
 
+    if (dev_data->extensions.vk_nv_framebuffer_mixed_samples) {
+        uint32_t raster_samples = static_cast<uint32_t>(GetNumSamples(pPipeline));
+        uint32_t subpass_color_samples = 0;
+
+        accumColorSamples(subpass_color_samples);
+
+        if (subpass_desc->pDepthStencilAttachment && subpass_desc->pDepthStencilAttachment->attachment != VK_ATTACHMENT_UNUSED) {
+            const auto attachment = subpass_desc->pDepthStencilAttachment->attachment;
+            const uint32_t subpass_depth_samples =
+                static_cast<uint32_t>(pPipeline->rp_state->createInfo.pAttachments[attachment].samples);
+
+            if (pPipeline->graphicsPipelineCI.pDepthStencilState) {
+                const bool ds_test_enabled = (pPipeline->graphicsPipelineCI.pDepthStencilState->depthTestEnable == VK_TRUE) ||
+                                             (pPipeline->graphicsPipelineCI.pDepthStencilState->depthBoundsTestEnable == VK_TRUE) ||
+                                             (pPipeline->graphicsPipelineCI.pDepthStencilState->stencilTestEnable == VK_TRUE);
+
+                if (ds_test_enabled && (!IsPowerOfTwo(subpass_depth_samples) || (raster_samples != subpass_depth_samples))) {
+                    skip |= log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_PIPELINE_EXT,
+                                    HandleToUint64(pPipeline->pipeline), "VUID-VkGraphicsPipelineCreateInfo-subpass-01411",
+                                    "vkCreateGraphicsPipelines: pCreateInfo[%d].pMultisampleState->rasterizationSamples (%u) "
+                                    "does not match the number of samples of the RenderPass depth attachment (%u).",
+                                    pipelineIndex, raster_samples, subpass_depth_samples);
+                }
+            }
+        }
+
+        if (IsPowerOfTwo(subpass_color_samples)) {
```

Not clear on the rational for the `IsPowerOfTwo` test here.  Clearly the parameter_validation layer should emit an error regarding for having more than one bit set in this (does it?), but it's not clear that one would want to silence these errors even in that case.

---

When removing the IsPowerOfTwo test I end up having more than one possible value for the number of samples that I use to test against raster_samples. To perform the tests I would need to select one of these values. I can use the max as you suggested in the initial review or use a kind of worst case and choose either min or max depending on the VUID.
VUID 01412: if (raster_samples < subpass_max_color_samples) 
VUID 01415: if ((raster_samples > subpass_min_color_samples) && ..
Can you please let me know your preference ?

---

If you can verify that the invalid case of having more than one possible value for the number of samples is caught elsewhere... then I'm fine with it as is.

---

Then okay as is.